pkgs/build-support/gradle/update-deps.nix

```diff
@@ -0,0 +1,154 @@
+{ lib
+, pkgs
+, stdenv
+, writeScript
+, writeShellScript
+, nix
+, gradle
+, openssl
+, mitm-cache
+, jdk
+, nsjail
+, path
+, python3
+}:
+
+# path to the derivation attr (usually the package name)
+{ attrPath ? pname
+# pname (attrPath synonym)
+, pname ? ""
+# the Nix code to get the derivation (this will be put in Bash double quotes)
+# this is relevant for downstream (non-nixpkgs) users
+, code ? "with import ./. { }; ($attrPath)"
+# unlike gradleFlags in normal packages, these flags are escaped so they are allowed to contain spaces
+# also, gradleFlags from configurePhase are used by default
+, gradleFlags ? []
+# attribute overrides for the "source" derivation
+, attrOverrides ? {}
+# run before fetching the dependencies (in sandbox)
+, preBuild ? ""
+# run before starting the sandbox for fetching dependencies
+, preBuildNoSandbox ? ""
+# the gradle commands for fetching the necessary dependencies
+, gradleCommands ? "gradle nixDownloadDeps"
+# run after fetching the dependencies (in sandbox)
+, postBuild ? ""
+# run after the sandbox for fetching dependencies has stopped
+, postBuildNoSandbox ? ""
+, mitmCachePort ? 1337
```

Comment:
Maybe we should rely on something like ephemeral-port-reserve, to find a free and unused port